TeachingRecordSystem/src/TeachingRecordSystem.Core/DataStore/Postgres/Models/ProfessionalStatus.cs

```diff
@@ -191,4 +191,49 @@ public void Update(
             Changes = changes,
         };
     }
+
+    public void Delete(
+        IReadOnlyCollection<RouteToProfessionalStatus> allRoutes,
+        string? deletionReason,
+        string? deletionReasonDetail,
+        EventModels.File? evidenceFile,
+        EventModels.RaisedByUserInfo deletedBy,
+        DateTime now,
+        out ProfessionalStatusDeletedEvent @event)
+    {
+        if (DeletedOn is not null)
+        {
+            throw new InvalidOperationException("Professional status is already deleted.");
+        }
+        if (Person is null)
+        {
+            throw new InvalidOperationException("Professional status is not linked to a person and cannot be deleted");
+        }
+
+        DeletedOn = now;
+        UpdatedOn = now;
+
+        var oldEventModel = EventModels.ProfessionalStatus.FromModel(this);
+        var oldPersonAttributes = EventModels.ProfessionalStatusPersonAttributes.FromModel(Person);
+        var oldRoute = allRoutes.Single(r => r.RouteToProfessionalStatusId == RouteToProfessionalStatusId);
+        var oldProfessionalStatusType = oldRoute.ProfessionalStatusType;
+
+        var route = allRoutes.Single(r => r.RouteToProfessionalStatusId == RouteToProfessionalStatusId);
+        var professionalStatusType = route.ProfessionalStatusType;
+        var personAttributesUpdated = Person.RefreshProfessionalStatusAttributesForDelete(professionalStatusType, allRoutes, this);
+
+        @event = new ProfessionalStatusDeletedEvent()
```

Comment:
Need a `Changes` member on here to track changes to person fields, like `ProfessionalStatusCreatedEvent`